### ISSUE TITLE
DefaultStorageRpc.read setting wrong range

### ISSUE THREAD

**Issue by mziccard** [GoogleCloudPlatform/gcloud-java#373]
In [DefaultStorageRpc.read](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java#L418) we set the content range as:
```java
public byte[] read(StorageObject from, Map<Option, ?> options, long position, int bytes)
    throws StorageException {
  ...
  downloader.setContentRange(position, (int) position + bytes);
  ...
}
```

Accorting to [here](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35) positions are inclusive, so we should probably do: `downloader.setContentRange(position, (int) position + bytes - 1);`

**Comment by aozarov** [comment]
Yes, I think you are right.

I think consequences of this bug are minor and basically turns our chunkSize max to be `chunkSize + 1 ` or in the "worse" case when the given `byteBuffer` is more than `chunkSize` we will be left with one extra byte to be consumed by itself in the next read).

Am I correct? Regardless, good catch and it should be fixed.

I am actually more concern though about google/google-api-java-client#937 as it will impact large downloads. (maybe we should send them a PR...)

**Comment by mziccard** [comment]
Yes exactly that was the problem I experienced when doing some tests for #359, a single byte was unexpectedly left to be read.

I am happy to send a PR to fix google/google-api-java-client#937
